ocaml-lsp-server/test/e2e-new/start_stop.ml

```diff
@@ -0,0 +1,128 @@
+open Stdune
+open Fiber.O
+
+let _PATH =
+  Bin.parse_path (Option.value ~default:"" @@ Env.get Env.initial "PATH")
+
+let bin = Bin.which "ocamllsp" ~path:_PATH |> Option.value_exn |> Path.to_string
+
+let env = Spawn.Env.of_list [ "OCAMLLSP_TEST=true" ]
+
+module Client = Lsp_fiber.Client
+open Lsp.Types
+
+let%expect_test "start/stop" =
+  let stdin_i, stdin_o = Unix.pipe ~cloexec:true () in
+  let stdout_i, stdout_o = Unix.pipe ~cloexec:true () in
+  let pid =
+    Spawn.spawn ~env ~prog:bin ~argv:[ bin ] ~stdin:stdin_i ~stdout:stdout_o ()
+  in
+  Unix.close stdin_i;
+  Unix.close stdout_o;
+  let handler = Client.Handler.make () in
+  let init =
+    let blockity =
+      if Sys.win32 then `Blocking
+      else (
+        Unix.set_nonblock stdout_i;
+        Unix.set_nonblock stdin_o;
+        `Non_blocking true)
+    in
+    let make fd what =
+      let fd = Lev_fiber.Fd.create fd blockity in
+      Lev_fiber.Io.create fd what
+    in
+    let* in_ = make stdout_i Input in
+    let* out = make stdin_o Output in
+    let io = Lsp_fiber.Fiber_io.make in_ out in
+    let client = Client.make handler io () in
+    let run_client () =
+      let capabilities = ClientCapabilities.create () in
+      Client.start client (InitializeParams.create ~capabilities ())
+    in
+    let print_init =
+      let+ resp = Client.initialized client in
+      print_endline "client: server initialized with:";
+      InitializeResult.yojson_of_t resp
+      |> Yojson.Safe.pretty_to_string ~std:false
+      |> print_endline
+    in
+    let run =
+      let* () = print_init in
+      print_endline "client: shutting down server";
+      Client.request client Shutdown
+    in
+    Fiber.fork_and_join_unit run_client (fun () ->
+        run
+        >>> Fiber.all_concurrently_unit
+              [ Lsp_fiber.Fiber_io.close io `Read
+              ; Lsp_fiber.Fiber_io.close io `Write
+              ; Client.stop client
+              ])
+  in
+  let waiting = ref true in
```

Comment:
`waiting` doesn't seem to be read anywhere?

Reply:
it's gone.